src/Compilers/CSharp/Portable/Syntax/GenericNameSyntax.cs

```diff
@@ -15,5 +16,12 @@ public bool IsUnboundGenericName
                 return this.TypeArgumentList.Arguments.Any(SyntaxKind.OmittedTypeArgument);
             }
         }
+
+        internal override string ErrorDisplayName()
+        {
+            var pb = PooledStringBuilder.GetInstance();
+            pb.Builder.Append(Identifier.ValueText).Append("<").Append(new string(',', Arity - 1)).Append(">");
```

Comment:
Because when we are unable to *resolve* the type name, the actual type arguments are irrelevant.